### PR TITLE
chore: Bump k8s to 1.29

### PR DESCRIPTION
**Pull request by davidjumani**
Bump k8s deps to v1.29
Ref: https://github.com/solo-io/gloo/issues/8978

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
 
-go 1.21
+go 1.22.2
 
 replace (
 	github.com/Sirupsen/logrus => github.com/sirupsen/logrus v1.4.2
 	github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309
+
+	// Although protoc-gen-openapi is not required in this repo, pin it with a compatible version anyway since this is used in other repos
+	github.com/getkin/kin-openapi => github.com/getkin/kin-openapi v0.107.0
```
Comment (sheidkamp) [inline go.mod:10]: Is this comment right? Don't we use a solo repository for protoc-gen-openapi? (github.com/solo-io/protoc-gen-openapi)

Comment (sheidkamp) [reply go.mod:10]: So if I understand correctly, this pin will make it so other repos that use github.com/solo-io/protoc-gen-openapi won't have to include the pin? And they will still be able to override this pin with a replace of its own?

Comment (davidjumani) [reply go.mod:10]: Pretty much - while this isn't necessary it just seems cleaner to keep all the modules in sync

**Review by bewebi — APPROVED** [review]
LGTM

**Review by sheidkamp — APPROVED** [review]

**Comment by davidjumani** [comment]
kick bulldozer

**Pull request merged**